src/kv/key.rs

```diff
@@ -48,6 +48,12 @@ impl<'k> Key<'k> {
     pub fn as_str(&self) -> &str {
         self.key
     }
+
+    /// Try get a string borrowed for the `'k` lifetime from this key.
+    pub fn to_borrowed_str(&self) -> Option<&'k str> {
```

Comment:
This returns an `Option` even though it's currently always `Some` so that we've got room to make the internals something like a `Cow<'k, str>`.